frontend/pages/hosts/details/OSSettingsModal/OSSettingsTable/OSSettingsTableConfig.tsx

```diff
@@ -53,6 +54,49 @@ interface IDataColumn {
   sortType?: string;
 }
 
+/**
+ * generates the formatted tooltip for the error column.
+ * the expected format of the error string is:
+ * "key1: value1, key2: value2, key3: value3"
+ */
+const generateFormattedTooltip = (detail: string) => {
+  const formattedText = detail.split(/[:,]+/).map((item, i, arr) => {
+    item.trim();
+    const key = `${item}-${i}`;
+    if (i % 2 === 0) {
+      return (
+        <>
+          <b key={key}>{item}: </b>
+        </>
+      );
+    }
+    return (
+      <span key={key}>
+        {item}
+        {/* dont include comma on the last item */}
+        {i === arr.length - 1 ? "" : ", "}
+      </span>
+    );
+  });
+
+  return <>{formattedText}</>;
+};
```

Comment:
just for fun, a slightly less efficient (because we have to split twice) but more concise version:

```ts
const generateFormattedTooltip = (detail: string) => {
  const keyValuePairs = detail.split(/, */);
  const formattedElements = keyValuePairs.map((pair, index) => {
    const [key, value] = pair.split(/: */);

    return (
      <span key={key}>
        <b>{key.trim()}:</b> {value.trim()}
        {index !== keyValuePairs.length - 1 && ", "}
      </span>
    );
  });

  return <>{formattedElements}</>;
};
```

Reply:
looks good. I prefer your suggestion because it reads easier.